Batching

mcdonnelldean opened the pull request:
@mcollina Please review?

mcdonnelldean committed:
First cut at batched

mcdonnelldean commented on srv/demo.js:
```diff
@@ -1,12 +1,30 @@
+    }
+  },
+  bole: {
+    level: 'debug',
```
This could actually just be info for the included demo

mcollina commented on influx-sink.js:
```diff
@@ -16,37 +23,102 @@ var defaults = {
+  locals.client = influx(locals.influx)
+
+  locals.batch.list = {}
+  locals.batch.stream = through.obj(on_write)
```
Instead of using through, you can just use a Writable and pass in a [https://nodejs.org/api/stream.html#stream_writable_writev_chunks_callback]() function (you still need to implement `_write` though), so that you do not have a timeout, but you write down to influx as soon as your batch is filled up.

mcdonnelldean replied:
Ah but then I loose the functionality of write to batch size x or every y msecs.

mcollina commented:
:+1:, that nit can be fixed later on if you like

mcdonnelldean committed:
See notes,
added port to demo to avoid port collisions
brought down default timeout and max
Updated deps and added toolbagrc
